trio/_subprocess.py

```diff
@@ -155,67 +242,718 @@ def __init__(
         self.args = self._proc.args
         self.pid = self._proc.pid
 
+    def __repr__(self):
+        if self.returncode is None:
+            status = "running with PID {}".format(self.pid)
+        else:
+            if self.returncode < 0:
+                status = "exited with signal {}".format(-self.returncode)
+            else:
+                status = "exited with status {}".format(self.returncode)
+            if self.returncode != 0 and not self.failed:
+                status += " (our fault)"
+        return "<trio.Process {!r}: {}>".format(self.args, status)
+
     @property
     def returncode(self):
-        """The exit status of the process (an integer), or ``None`` if it has
-        not exited.
+        """The exit status of the process (an integer), or ``None`` if it is
+        not yet known to have exited.
+
+        By convention, a return code of zero indicates success.  On
+        UNIX, negative values indicate termination due to a signal,
+        e.g., -11 if terminated by signal 11 (``SIGSEGV``).  On
+        Windows, a process that exits due to a call to
+        :meth:`Process.terminate` will have an exit status of 1.
 
-        Negative values indicate termination due to a signal (on UNIX only).
-        Like :attr:`subprocess.Popen.returncode`, this is not updated outside
-        of a call to :meth:`wait` or :meth:`poll`.
+        Accessing this attribute does not check for termination; use
+        :meth:`poll` or :meth:`wait` for that.
         """
         return self._proc.returncode
 
+    @property
+    def failed(self):
+        """Whether this process should be considered to have failed.
+
+        If the process hasn't exited yet, this is None. Otherwise, it is False
```

Comment:
This seems tricky and error-prone (having it be True, False or None)

Reply:
I feel like this is the convention I usually see for "optional bool", and it mirrors the `returncode` attribute (which is None or an integer). Would you prefer that accessing `.failed` throw an exception if the process is still running?